src/AutoMapper/Configuration/Annotations/AutoMapAttribute.cs

```diff
@@ -81,6 +86,11 @@ public void ApplyConfiguration(IMappingExpression mappingExpression)
             {
                 mappingExpression.ConvertUsing(TypeConverter);
             }
+
+            if (AsProxy)
```

Comment:
Here could be a check if Dest type is interface or not

Reply:
That would be the job of the fluent API. So no need.